Source/DafnyCore/Verifier/ProofObligationDescription.cs

```diff
@@ -5,6 +5,14 @@
 namespace Microsoft.Dafny.ProofObligationDescription;
 
 public abstract class ProofObligationDescription : Boogie.ProofObligationDescription {
+  public abstract Expression GetAssertedExpr(DafnyOptions options);
+}
+
+// When there is no way to translate the asserted constraint in Dafny
+public abstract class ProofObligationDescriptionNotInDafny : ProofObligationDescription {
```

Comment:
What do you think of this name, instead?
```suggestion
public abstract class ProofObligationDescriptionWithNoExpr : ProofObligationDescription {
```